tests/integration/it_test_authorization.py

```diff
@@ -0,0 +1,111 @@
+import random
+import string
+from datetime import datetime
+
+import flask
+import pytest
+
+from timeless.access_control.authorization import is_allowed
+from timeless.access_control.methods import Method
+from timeless.companies.models import Company
+from timeless.employees.models import Employee
+
+
+# @todo #329:30min After #298 is finished update this test cases with
+#  appropriate roles and make sure that current test cases diffrentiate enough
+#  between roles, so we can be sure that is_allowed takes into account all
+#  exiting roles. Also remove pytest.mark.skip from all test methods here.
+def test_manager_can_access_employee(app, db_session):
+    my_company = Company(
+        name="Acme Inc.", code="code1", address="addr"
+    )
+    db_session.add(my_company)
+    db_session.commit()
+    me = create_employee(my_company, "manager")
+    db_session.add(me)
+    flask.g.user = me
+    other = create_employee(my_company, "master")
+    db_session.add(other)
+    db_session.commit()
+    assert is_allowed(
+        method=Method.READ, resource="employee", employee_id=other.id
+    )
+
+
+def test_director_can_access_manager(app, db_session):
+    my_company = Company(
+        name="Acme Inc.", code="code1", address="addr"
+    )
+    db_session.add(my_company)
+    db_session.commit()
+    me = create_employee(my_company, "director")
+    db_session.add(me)
+    flask.g.user = me
+    other = create_employee(my_company, "manager")
+    db_session.add(other)
+    db_session.commit()
+    assert is_allowed(
+        method=Method.READ, resource="employee", employee_id=other.id
+    )
+
+
+@pytest.mark.skip()
+def test_administrator_can_access_director_from_other_company(app, db_session):
+    my_company = Company(
+        name="Acme Inc.", code="code1", address="addr"
+    )
+    db_session.add(my_company)
+    db_session.commit()
+    me = create_employee(my_company, "administrator")
+    db_session.add(me)
+    flask.g.user = me
+    other_company= Company(
+        name="Foo Inc.", code="code1", address="addr"
+    )
+    db_session.add(other_company)
+    other = create_employee(other_company, "director")
+    db_session.add(other)
+    db_session.commit()
+    assert is_allowed(
+        method=Method.READ, resource="employee", employee_id=other.id
+    )
+
+
+def test_owner_can_access_director(app, db_session):
```

Comment:
@krzyk all these tests have the same code. Could we refactor it with using `pytest.mark.parametrize` ? I think it will look more pretty.

Reply:
@vladarefiev I don't think so, current tests don't use roles yet, but after migration one would first need to create 2 roles, commit them to DB and only then use those. I don't see how it would be possible using parameters

Reply:
@krzyk For example:
```
@pytest.mark.parametrize("my_role,other_role", (
    ("manager", "master"),
    ("director", "manager")
))
def test_access(my_role, other_role, app, db_session):
    my_company = Company(
        name="Acme Inc.", code="code1", address="addr"
    )
    db_session.add(my_company)
    db_session.commit()
    me = create_employee(my_company, my_role)
    db_session.add(me)
    flask.g.user = me
    other = create_employee(my_company, other_role)
    db_session.add(other)
    db_session.commit()
    assert is_allowed(
        method=Method.READ, resource="employee", employee_id=other.id
    )
```
instead of both `test_manager_can_access_employee` and `test_director_can_access_manager`

Reply:
@vladarefiev Might be possible, but with added creation of roles inside the test. I added this to the todo, so when we will have roles here it should be easy to decide if it is doable.